exporter/exporterqueue/queue.go

```diff
@@ -0,0 +1,69 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package exporterqueue // import "go.opentelemetry.io/collector/exporter/exporterqueue"
+
+import (
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/exporter"
+	"go.opentelemetry.io/collector/exporter/internal"
+)
+
+// Queue defines a producer-consumer exchange which can be backed by e.g. the memory-based ring buffer queue
+// (boundedMemoryQueue) or via a disk-based queue (persistentQueue)
+// This API is at the early stage of development and may change without backward compatibility
+// until https://github.com/open-telemetry/opentelemetry-collector/issues/8122 is resolved.
+type Queue[T any] internal.Queue[T]
+
+// Settings defines settings for creating a queue.
+type Settings[T any] struct {
+	Sizer            internal.Sizer[T]
```

Comment:
Anti pattern to expose internal types like this. Consider to move the definition of Sizer at least.

Also, is the sizer/type not defined by the config arguments? Why do you need to pass it here?

Reply:
Removed this field 👍